user-interface/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/EditBatchDialog.java

```diff
@@ -0,0 +1,228 @@
+package life.qbic.datamanager.views.projects.project.samples.registration.batch;
+
+import com.vaadin.flow.component.AbstractField.ComponentValueChangeEvent;
+import com.vaadin.flow.component.ClickEvent;
+import com.vaadin.flow.component.ComponentEvent;
+import com.vaadin.flow.component.ComponentEventListener;
+import com.vaadin.flow.component.button.Button;
+import com.vaadin.flow.component.html.Div;
+import com.vaadin.flow.component.html.Span;
+import com.vaadin.flow.component.textfield.TextField;
+import java.util.Collections;
+import java.util.List;
+import life.qbic.datamanager.views.general.DialogWindow;
+import life.qbic.datamanager.views.projects.project.samples.registration.batch.EditBatchDialog.ConfirmEvent.Data;
+import life.qbic.datamanager.views.projects.project.samples.registration.batch.SampleBatchInformationSpreadsheet.SampleInfo;
+import life.qbic.projectmanagement.domain.model.batch.BatchId;
+import life.qbic.projectmanagement.domain.model.experiment.ExperimentalGroup;
+import life.qbic.projectmanagement.domain.model.experiment.vocabulary.Analyte;
+import life.qbic.projectmanagement.domain.model.experiment.vocabulary.Species;
+import life.qbic.projectmanagement.domain.model.experiment.vocabulary.Specimen;
+
+/**
+ * A dialog used for sample batch registration.
+ */
+public class EditBatchDialog extends DialogWindow {
+
+  private final SampleBatchInformationSpreadsheet spreadsheet;
+  private final BatchId batchId;
+  private final List<SampleInfo> existingSamples;
+
+  private final TextField batchNameField;
+  private final Span batchNameText;
+
+  public EditBatchDialog(String experimentName,
+      List<Species> species,
+      List<Specimen> specimen,
+      List<Analyte> analytes,
+      List<ExperimentalGroup> experimentalGroups,
+      BatchId batchId,
+      String batchName,
+      List<SampleInfo> existingSamples) {
+
+    addClassName("batch-update-dialog");
+    setConfirmButtonLabel("Update Samples");
+
+    this.batchId = batchId;
+    this.existingSamples = Collections.unmodifiableList(existingSamples);
+
+    spreadsheet = new SampleBatchInformationSpreadsheet(experimentalGroups, species, specimen,
+        analytes, false);
+
+    batchNameField = new TextField();
+    batchNameField.addClassName("batch-name-field");
+    batchNameField.setLabel("Batch Name");
+    batchNameField.setPlaceholder("Please enter a name for this batch");
+    batchNameField.setRequired(true);
+    // must contain at least one non-whitespace character and no leading/tailing whitespace.
+    batchNameField.setPattern("^\\S+(.*\\S)*$");
+    batchNameField.setErrorMessage(
+        "The batch name must not be empty. It must not start nor end with whitespace.");
+    batchNameField.setValue(batchName);
```

Comment:
As SF also mentioned in the coding Kata i would strongly urge you to move the component styling and adaption into seperate methods so the constructor is not bloated